fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -236,12 +239,16 @@ private boolean insertField(LuceneDocumentFromRecord.DocumentField field, final
 
     private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
                                byte[] primaryKey) throws IOException {
-        final IndexWriter newWriter = directoryManager.getIndexWriter(groupingKey, indexAnalyzer);
+        final List<String> texts = fields.stream()
+                .filter(f -> f.getType().equals(LuceneIndexExpressions.DocumentFieldType.TEXT))
+                .map(f -> (String) f.getValue()).collect(Collectors.toList());
+        final IndexWriter newWriter = directoryManager.getIndexWriter(groupingKey,
+                indexAnalyzerChooser.chooseAnalyzer(texts));
         BytesRef ref = new BytesRef(primaryKey);
         Document document = new Document();
         document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
         document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
-        final AnalyzingInfixSuggester suggester = autoCompleteEnabled ? getSuggester(groupingKey) : null;
+        final AnalyzingInfixSuggester suggester = autoCompleteEnabled ? getSuggester(groupingKey, texts) : null;
```

Comment:
I think the `FDBDirectoryWrapper` also caches the `AnalyzingInfixSuggester`, so this also returns an `AnalyzingInfixSuggester` with the first analyzer it receives in a transaction

Reply:
Fixed it